### DISCUSSION TITLE
How to add transitions between backgrounds

### DISCUSSION THREAD

**Discussion by Azurysu** [dialogic-godot/dialogic#2317] [Q&A]
Hello,
I want to add a fade transition between backgrounds whenever they change for a visual novel I'm working on, but so far I haven't made any progress on that. Dialogic doesn't have it as an option in the timeline editor, and since I'm new to this plugin I'm not quite sure at what should I be looking for to change it.

I've been reading the documentation and I know there's built-in stuff for background transitions, but I don't know how to implement them. I think it's done through 'subsystem_Background.gd' and emiting a dialogic 'background_changed' signal whenever I want the BG to make the fade transition, but I don't know how to edit it so the signal actually works.

Thank you in advance and sorry if I haven't explained myself properly; english is not my first language.

**Comment by Jowan-Spooner** [comment]
The background fade and transition is configurable on the background event, without any coding. What version are you using?

**Reply by Azurysu** [reply]
Ah my bad, I'm sorry for not saying it in the main message. I'm using Dialogic 2.0, the latest alpha (n°14 I think). 

I've tried configuring that event as you say and found there's different transitions on the visual editor (that's on me for not paying attention) but so far I only got the first background transition to work. Any other bg changes don't seem to work, only the first background in the timeline does.

Thank you so much for your help and your patience. I attach a video and a screenshot of the code in the visual editor.



https://github.com/dialogic-godot/dialogic/assets/114650730/f2a6f03e-ce14-43c3-a07f-52a81e397f35


![imagen_2024-06-28_142045665](https://github.com/dialogic-godot/dialogic/assets/114650730/992baf3e-7bcc-491d-85dc-b4b62ffd6c94)

**Reply by Jowan-Spooner** [reply]
That's strange, maybe it's a bug. I can check on Monday.
